resources/idlharness.js

```diff
@@ -218,6 +226,81 @@ IdlArray.prototype.add_untested_idls = function(raw_idls, options)
     this.internal_add_idls(parsed_idls, options);
 };
 
+//@}
+IdlArray.prototype.add_dependency_idls = function(raw_idls, options)
+//@{
+{
+    const parsed_idls = WebIDL2.parse(raw_idls);
+    const new_options = { only: [] }
+    
+    const all_deps = new Set();
+    Object.values(this["inheritance"]).forEach(v => all_deps.add(v));
+    Object.values(this["implements"]).forEach(v => all_deps.add(v));
+    this.partials.map(p => p.name).forEach(v => all_deps.add(v));
+    
+    console.log('all deps', all_deps);
+
+    if (options && options.except && options.only) {
+        throw new IdlHarnessError("The only and except options can't be used together.");
+    }
+    const skip = name => {
+        return !name
+            || options && (
+                options.except && options.except.includes(name)
+                || options.only && !options.only.includes(name));
+    }
+
+    // Record of skipped items in case we later determine they are a dependency.
+    const skipped = new Map();
+    const process = function(parsed) {
+        let name = parsed.name
+            || parsed.type == "implements" && parsed.target
+            || parsed.type == "includes" && parsed.target;
+        if (skip(name) || !all_deps.has(name)) {
+            name && skipped.set(name, parsed);
+            return;
+        }
+
+        console.log(parsed);
+        
+        new_options.only.push(name);
+        const follow_up = [];
```

Comment:
Can you add a comment about what this bookkeeping is for? I think it's in case we have a dependency on B, which in turn depends on C, but we've already processed C. However, since `add_dependency_idls` can be called multiple times, once for each IDL file, it seems like this will still be broken for cross-spec dependencies. Most likely, some spec depends on something in HTML, that depends on something in DOM, but we process DOM before HTML. Does this still work then?

Reply:
I agree, something more global would have its own set of problems. ("Why does removing the *following* line cause a problem with the IDL on *this* line?")

The "Record of skipped items in case we later determine they are a dependency" comment is clear to me now. Could possibly say that it's *not* global, but that's a pretty far-fetched misunderstanding, so nah.